EspionSpotify.Updater/Program.cs

```diff
@@ -0,0 +1,137 @@
+﻿using EspionSpotify.Updater.GitHub;
+using Ionic.Zip;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace EspionSpotify.Updater
+{
+    class Program
+    {
+        static void Main()
+        {
+            Thread t = new Thread(() =>
+            {
+                ProcessUpdate();
+            });
+            t.Start();
+        }
+        public static void ProcessUpdate()
+        {
+            Console.WriteLine("Downloading update...");
+            if (!downloadUpdate())
+            {
+                Console.WriteLine("Press any key to exit");
+                Console.ReadKey();
+                Environment.Exit(1);
+            }
+            try
+            {
+                Console.WriteLine("Extracting");
+                extract(AppDomain.CurrentDomain.BaseDirectory);
+            }
+            catch (System.UnauthorizedAccessException ex)
+            {
+                Console.WriteLine("An error occurred while extracting. Make sure Spytify is not running or Antivirus Software is interferring.\n"+
+                    "You can extract the file manually at: "+ AppDomain.CurrentDomain.BaseDirectory+"../update.zip\n"+
+                    "Error Message: "+ ex.Message);
+                Console.WriteLine("Press any key to exit");
+                Console.ReadKey();
+                Environment.Exit(1);
+            }
+            catch(Exception ex)
+            {
+                Console.WriteLine("An error occurred while extracting\n" +
+                    "The error was" + ex.Message);
+                Console.ReadKey();
+            }
+            File.Delete(AppDomain.CurrentDomain.BaseDirectory + "../update.zip");
+            Console.WriteLine("Update successfull");
+            Process.Start(new ProcessStartInfo(AppDomain.CurrentDomain.BaseDirectory + "../Spytify.exe", "upgradeSettings"));
+            Environment.Exit(0);
+        }
+        public static bool isVersionNewer(string name)
+        {
+            Match m = Regex.Match(name, @"v\.((?:\d+\.)*\d+?).zip$");
+            if (m.Groups[1].Success)
+            {
+                string currentVersion = null;
+                try
+                {
+                    currentVersion = FileVersionInfo.GetVersionInfo(AppDomain.CurrentDomain.BaseDirectory + "../Spytify.exe").FileVersion;
+                }
+                catch
+                {
+                    return true;
+                }
+                
+                if (currentVersion.StartsWith(m.Groups[1].Value))
```

Comment:
official download should always start with spytify and end with zip "Spytify-v1.8.1.zip", so we want to ignore those:
`ADVERTISEMENT-ONLY-RECORDER-VERSION.-Spytify-v.1.8.zip`
`Spytify-v.1.8-debug.zip`
and version will always have major, but other are optional, so what I mean is:
`v.MAJOR.MINOR.PATCH` -- 4th number BUILD is always ignored, so it won't appear
so maybe a strict regex test will be better

```
var r = new System.Text.RegularExpressions.Regex(@"^Spytify(-|\.)(v?)(\.)?\d+(\.\d+)?(\.\d+)?.zip$", System.Text.RegularExpressions.RegexOptions.IgnoreCase);

Console.WriteLine(r.IsMatch("ADVERTISEMENT-ONLY-RECORDER-VERSION.-Spytify-v.1.8.zip")); // false
Console.WriteLine(r.IsMatch("Spytify-v.1.8-debug.zip")); // false
Console.WriteLine(r.IsMatch("Spytify.va.b.zip")); // false

Console.WriteLine(r.IsMatch("Spytify.v1.zip")); // true - with dot version

Console.WriteLine(r.IsMatch("Spytify-v1.zip")); // true - with dash version no-dot

Console.WriteLine(r.IsMatch("Spytify-v.1.zip")); // true - major - default (dash version dot)
Console.WriteLine(r.IsMatch("Spytify-v.1.8.zip")); // true - major minor
Console.WriteLine(r.IsMatch("Spytify-v.1.8.1.zip")); // true - major minor patch

Console.WriteLine(r.IsMatch("spytify-v.1.zip")); // true - lowercase
```